spec/commands/add_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+describe "bundle add" do
+  before :each do
+    build_repo2
+
+    gemfile <<-G
+      source "file://#{gem_repo2}"
+    G
+  end
+
+  context "when version number is set" do
+    it "adds gem with provided version" do
+      bundle "add activesupport 2.3.5"
+      expect(bundled_app("Gemfile").read).to include("gem 'activesupport', '~> 2.3.5'")
+    end
+
+    it "adds gem with provided version and version operator" do
+      update_repo2 do
+        build_gem "activesupport", "3.0"
+      end
+
+      bundle "add activesupport '> 2.3.5'"
+      expect(bundled_app("Gemfile").read).to include("gem 'activesupport', '> 2.3.5'")
+    end
+  end
+
+  context "when version number is not set" do
+    it "adds gem with last stable version" do
+      bundle "add activesupport"
+      expect(bundled_app("Gemfile").read).to include("gem 'activesupport', '~> 2.3.5'")
+    end
+
+    it "adds the gem with the last prerelease version" do
+      update_repo2 do
+        build_gem "activesupport", "3.0.0.beta"
+      end
+
+      bundle "add activesupport --pre"
+      expect(bundled_app("Gemfile").read).to include("gem 'activesupport', '~> 3.0.0.beta'")
+    end
+  end
+
+  context "when group is set" do
+    it "adds the gem with the specified groups" do
+      bundle "add activesupport --group development test"
+      expect(bundled_app("Gemfile").read).to include("gem 'activesupport', '~> 2.3.5', :group => [:development, :test]")
+    end
+  end
+
+  context "when source is set" do
```

Comment:
Let's also have a spec or two for multiple flag options (ex. specifying `bundle add --group development test --source https://proxy.org`)